### ISSUE TITLE
Markers in Global map is not work with active plugin Polylang

### ISSUE THREAD

**Issue by GoogleCodeExporter** [cyberhobo/wordpress-geo-mashup#661]
```
Geo Mashup 1.6.2
Polylang 1.1.2

If Polylang active than global map is show just map, without markers. Problem 
with languages.
```

Original issue reported on code.google.com by `istomin....@gmail.com` on 16 Jul 2013 at 7:11

**Comment by flknx** [comment]
Hi,

Just stumbled to this problem, I think (atleast the global map isn't loading :). Unfortunately, I can't really JS, but if anyone's interested in this problem, here's the error that appears on page with global map.

TypeError: Cannot read property 'color' of undefined
#5 wp-content/plugins/geo-mashup/js/taxonomy.js?ver=1.8.4:413:50
#4 wp-content/plugins/geo-mashup/js/taxonomy.js?ver=1.8.4:413:50 Object.jQuery.extend.term_manager.term_manager.extendTerm()
#3 wp-content/plugins/geo-mashup/js/geo-mashup.js?ver=1.8.4:708:31 Object.GeoMashup.addObjects()
#2 wp-content/plugins/geo-mashup/js/geo-mashup-mxn.js?ver=1.8.4:743:8 Object.GeoMashup.createMap()
#1 ?geo_mashup_content=render-map&map_data_key=fb481e522fe732bff6e2e0589dfa4275&lang=fi&object_id=871&map_content=global&name=gm-map-2:39:11

However, in my use case, I was able to achieve pretty much the same end result with manually adding categories, for example [geo_mashup_map map_cat="9,11"], instead of using global maps option.

**Comment by vinylroads** [comment]
Hello,

I like geomashup. A great plugin which can do everything !

I have the same issue. Geomashup works very well but i can’t display a map if i activate translation with polylang.

Is it solved or do i need to use an other map plugin ?

I’ve tested a lot of thing and could not find a solution.

In console, i have this error :

`TypeError: undefined is not an object (evaluating 'term_properties[taxonomy].terms[term_id].color')`

Here is how i display map in template: 

`<?php echo GeoMashup::map('map_content=global&map_cat=planete') ?>`

The map is ok in a language (FR) but if i translate a post it doesn’t work anymore.

Best regards,

Sébastien

**Comment by cyberhobo** [comment]
@vinylroads , if you can give me a URL where I can see the issue that would help me at least determine a next step. Alternately, posting the arguments to the `GeoMashup.createMap( ... )` call in the map frame here would be useful.

**Comment by birtakimseyler** [comment]
Hello,
@cyberhobo @vinylroads 
Have you found any solution for Polylang issue? There's a problem with global map. Contextual map is working well.

**Comment by vinylroads** [comment]
Sorry, on this project that required polylang, i gave up and built my own custom map with custom type + ACF.

**Comment by eliazu** [comment]
i've edited the wp-content/plugins/geo-mashup/js/taxonomy.js file.

changed at line 413
`if ( term_properties[taxonomy].terms[term_id].color ) {
					color_name = term_properties[taxonomy].terms[term_id].color;
				} else {
					color_name = GeoMashup.color_names[ loaded_taxonomy.term_count % GeoMashup.color_names.length ];
				}`

to

`color_name = GeoMashup.color_names[ loaded_taxonomy.term_count % GeoMashup.color_names.length ];`

and removed the if statement at line 468

`if ( term_properties[taxonomy].terms[term_id].line_zoom ) {
					max_line_zoom = term_properties[taxonomy].terms[term_id].line_zoom;
				}`

of course i did this in the min.js file ...

after this changes the map works with polylang!

**Comment by eliazu** [comment]
@Birtakimseyler it works for me.  for all languages.
the English page, with code that query X last posts, queries only English places.
the french page,  with code that query X last posts, queries only french places.

maybe you put the code not in a page with language defined, or you queried wrong posts languages or undefined languages.

**Comment by Birtakimseyler** [comment]
@eliazu Yes you're right. It could be solved in the the queries. I did not used like that. But still there's a problem with "Nearby list". Did you already found another solution about that?

Thank you!
